pkg/demoinfocs/game_events.go

```diff
@@ -340,20 +340,39 @@ func (geh gameEventHandler) playerDeath(data map[string]*msg.CSVCMsg_GameEventKe
 }
 
 func (geh gameEventHandler) playerHurt(data map[string]*msg.CSVCMsg_GameEventKeyT) {
+	userID := data["userid"].GetValShort()
+	player := geh.playerByUserID32(userID)
 	attacker := geh.playerByUserID32(data["attacker"].GetValShort())
+
 	wepType := common.MapEquipment(data["weapon"].GetValString())
-	userID := data["userid"].GetValShort()
 	wepType = geh.attackerWeaponType(wepType, userID)
 
+	health := int(data["health"].GetValByte())
+	armor := int(data["armor"].GetValByte())
+	healthDamageTaken := health
```

Comment:
hmm, I think this is wrong. shouldn't it be `int(data["dmg_health"].GetValShort())` by default and if `data["health"].GetValByte() == 0` then it should be `player.Health()`.

but it should never be `data["health"].GetValByte()` - right?

Reply:
Yes, you're right ! My bad